var/spack/repos/builtin/packages/py-cartopy/package.py

```diff
@@ -21,7 +22,8 @@ class PyCartopy(PythonPackage):
     depends_on('py-pyshp@1.1.4:',   type=('build', 'run'))
     depends_on('py-six@1.3.0:',     type=('build', 'run'))
     depends_on('geos@3.3.3:')
-    depends_on('proj@4.9.0:5')
+    depends_on('proj@4.9.0:5', when='@0.16.0')
+    depends_on('proj@6:',      when='@0.17.0')
 
     # optional dependecies
     depends_on('py-matplotlib@1.5.1:', type=('build', 'run'))
```

Comment:
I would add variants for these optional dependencies. See the `requirements` directory in the tarball for variant names and which dependencies to add for each variant. Also, `py-filelock` is now needed for testing as well.

Reply:
@adamjstewart done. I'm not sure if I did `when` option for `py-pillow`: it is required for both `plotting` and `ows`.